src/context/AppContext.tsx

```diff
@@ -94,23 +97,46 @@ export const AppContextProvider = ({ children }: { children: ReactNode }) => {
     }
   }, [updateUser, wallet]);
 
+  const asyncFunction = useAsyncFunction({
+    submit: ActionTypes.WALLET_OPEN,
+    error: ActionTypes.WALLET_OPEN_ERROR,
+    success: ActionTypes.WALLET_OPEN_SUCCESS,
+  });
+
+  const connectWallet = useCallback(async () => {
+    setWalletConnecting?.(true);
+    let walletConnectSuccess = false;
+    try {
+      await asyncFunction(undefined);
+      walletConnectSuccess = true;
+    } catch (error) {
+      console.error('Could not connect wallet', error);
+    }
+    if (updateWallet && walletConnectSuccess) {
+      updateWallet();
+    }
+    setWalletConnecting?.(false);
+  }, [asyncFunction, updateWallet, setWalletConnecting]);
+
```

Comment:
Moved this here to allow it to be used by other components, like I have done within the `Join this Colony` button. Allowing us to prompt a user to connect, as opposed to direct them to the Colony creation page.

I feel this prompt is more relevant considering the user creation happens on connection now.

Reply:
> user creation happens on connection now

Not sure what you mean here?

Reply:
Not as far as I'm aware. We have the username and email inputs in the user creation flow so I don't think we could implement this (how would we select a username for the wallet, for instance?). Users are created when you run the temp script, maybe that's what's happening?

Reply:
Perhaps a misunderstanding, this is based off a conversation with Raul, which I thought I recall him saying that this was how it was setup.

Reply:
@rdig Perhaps you're able to clarify here.

Reply:
@willm30 Either way, I still feel it makes sense to extract it.

Reply:
Will is correct here. We did in fact had a conversation with Arren about this, but the user creation thing was merely suggested, not implemented.

It can be theoretically done, however it's not really strait forward and will require some work to implemenent.

Eventually we might want to have it, if we decide for it, but for the time being it's not going to be implemented.